### PR TITLE
Add a URL redirector for Google sheets (#342)

### PR DESCRIPTION
**Pull request by rawxfish**
This makes /puzzles/sheets/<puzzle_pk> direct to the sheets, or the home page if puzzle or sheets does not exist. I wasn't sure how to integrated it with the react or discord bots though.

### CODE REVIEW

File: puzzles/views.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+from puzzles.models import Puzzle
+
+
+@login_required(login_url="/")
+def redirect_to_sheet(request, puzzle_pk):
+    try:
+        puzzle = Puzzle.objects.get(pk=puzzle_pk)
+    except ObjectDoesNotExist:
+        messages.error(
```
Comment (asdfryan) [inline puzzles/views.py:17]: This is also testable right?

Comment (rawxfish) [reply puzzles/views.py:17]: true...

**Review by akirabaruah — CHANGES_REQUESTED** [review]
Overall, LGTM - thanks for adding this, @rawxfish

File: puzzles/urls.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from django.urls import path
+from .views import redirect_to_sheet
+
+app_name = "puzzles"
+urlpatterns = [
+    path("sheets/<int:puzzle_pk>", redirect_to_sheet, name="puzzle_sheet_redirect"),
```
[the remark this answers is not in the document]
Comment (rawxfish) [reply puzzles/urls.py:6]: will change it to `s/<pk>`

File: puzzles/views.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        )
+        return redirect("/")
+
+    if puzzle.sheet:
+        return HttpResponseRedirect(puzzle.sheet)
+    else:
+        messages.error(
+            request,
+            f"Sheets does not exist yet for {puzzle.name}.\n"
+            f"Please wait a few minutes. If you have been waiting for a while, please ping @devs",
```
[the remark this answers is not in the document]
Comment (rawxfish) [reply puzzles/views.py]: I made it a DISCORD_DEVS_ROLE env var that defaults to "dev"

**Review by asdfryan — APPROVED** [review]

**Review by akirabaruah — APPROVED** [review]
awesomesauce :+1: thanks!

**Review by rgossiaux — APPROVED** [review]
yay!

**Pull request merged**